logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -66,7 +70,43 @@ class ConversationDataSource(
         conversationDAO.getAllConversations().map { it.map(conversationMapper::fromDaoModel) }
     }
 
+    /**
+     * Gets a flow that allows observing of
+     */
+    override suspend fun getConversationDetailsById(conversationID: ConversationId): Flow<ConversationDetails> =
+        conversationDAO.getConversationByQualifiedID(idMapper.toDaoModel(conversationID))
+            .filterNotNull()
+            .map(conversationMapper::fromDaoModel)
+            .flatMapLatest(::getDetailsFlowConversation)
+
+    private suspend fun getDetailsFlowConversation(conversation: Conversation): Flow<ConversationDetails> =
+        when (conversation.type) {
+            Conversation.Type.SELF -> flowOf(ConversationDetails.Self(conversation))
+            Conversation.Type.GROUP -> flowOf(ConversationDetails.Group(conversation))
+            Conversation.Type.ONE_ON_ONE -> {
+                suspending {
+                    val selfUserId = userRepository.getSelfUser().map { it.id }.first()
+                    getConversationMembers(conversation.id).map { members ->
+                        members.first { itemId -> itemId != selfUserId }
+                    }.coFold({
+                        // TODO: How to Handle failure when dealing with flows?
```

Comment:
So from what I've read we throw exceptions and they can then we caught by the consumer and show an error message.

But in this case it would only throw this error if we can't read from the database or the member data is corrupted for some reason. So perhaps is fine to throw an `IOException` let the application crash if it happens?